core/src/main/java/software/amazon/awssdk/auth/profile/ProfileFile.java

```diff
@@ -0,0 +1,348 @@
+     * that the parents occur earlier than the children.
+     *
+     * @param unsorted The unsorted collection of profiles so far.
+     * @param sorted The sorted collection of profiles so far.
+     * @param profileName The name of the profile to move to the sorted collection.
+     * @param children Any known children of this profile, which will allow us to ensure there is no circular references in
+     * the configuration file.
+     */
+    private void sortProfileAndParents(Map<String, Map<String, String>> unsorted,
+                                       Map<String, Map<String, String>> sorted,
+                                       String profileName,
+                                       Set<String> children) {
+        Validate.validState(!children.contains(profileName),
+                            "Invalid profile file: Circular relationship detected with profiles %s.", children);
+
+        Map<String, String> profileProperties = unsorted.get(profileName);
+        Validate.validState(profileProperties != null, "Parent profile '%s' does not exist.", profileName);
+
+        String parentProfileName = profileProperties.get(ProfileProperties.SOURCE_PROFILE);
+
+        if (parentProfileName != null && !sorted.containsKey(parentProfileName)) {
+            // The parent hasn't been moved over yet. Move it over before we move this one over.
+            children.add(profileName);
+            sortProfileAndParents(unsorted, sorted, parentProfileName, children);
+            children.remove(profileName);
+        }
+
+        // The parents have all been moved to the sorted list, move this one over.
+        sorted.put(profileName, profileProperties);
+        unsorted.remove(profileName);
+    }
+
+    /**
+     * Convert the sorted map of profile properties into a sorted list of profiles.
+     */
+    private Map<String, Profile> convertToProfilesMap(Map<String, Map<String, String>> sortedProfiles) {
+        Map<String, Profile> result = new LinkedHashMap<>();
+        for (Entry<String, Map<String, String>> rawProfile : sortedProfiles.entrySet()) {
+            String parentProfileName = rawProfile.getValue().get(ProfileProperties.SOURCE_PROFILE);
+            Profile parentProfile = result.get(parentProfileName);
+
+            Profile profile = Profile.builder()
+                                     .name(rawProfile.getKey())
+                                     .properties(rawProfile.getValue())
+                                     .credentialsParent(parentProfile)
+                                     .build();
+
+            result.put(profile.name(), profile);
+        }
+
+        return result;
+    }
+
+    /**
+     * The supported types of profile files. The type of profile determines the way in which it is parsed.
+     */
+    public enum Type {
+        /**
+         * A configuration profile file, typically located at ~/.aws/config, that expects all profile names (except the default
```

Comment:
expects -> requires?

Or just some sort of more clarification?

Reply:
I'll clarify.